### PR TITLE
Scan: Tx msg restructure

### LINKED ISSUE #2637: Flesh out the new structure of `Msg.re`.

### LINKED ISSUE #2698: Implement Token msgs on `Msg.re`

### PR DESCRIPTION
**Pull request by pzshine**
Fixed: #2637 
Fixed: #2698 
Fixed: #2723 

## Implementation details

- Refactor on `Msg.re`
- Made two Msg to be used as example
  - SendMsg
  - ReceiveMsg

**Note: If you pick the continue task**

- Please `checkout -b` from this branch
- Now there are separate Msg categories components as this image below.

image

So I've already made the function for getting the badge color by each category on `MsgBadge.re`

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Review by Pitchanai — APPROVED** [review]

File: scan/src/components/TxIndexPageTable.re [hunk 1]
```diff
@@ -1021,37 +1036,40 @@ let make = (~messages: list(TxSub.Msg.t)) => {
          <TBody.Grid key={index |> string_of_int} paddingH={`px(24)}>
            <Row.Grid alignItems=Row.Start>
              <Col.Grid col=Col.Two>
-               <div className=Styles.badgeContainer>
-                 <div className={Styles.badge(theme.bgColor)}>
-                   <Text value={theme.text} size=Text.Xs color={theme.textColor} />
+               //TODO: will remove when doing the ravamp style
```
Comment (evilpeach) [inline scan/src/components/TxIndexPageTable.re]: revamp

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: CHANGELOG_UNRELEASED.md [hunk 2]
```diff
@@ -27,6 +27,8 @@
 
 ### Scan
 
+- (impv) [\#2723](https://github.com/bandprotocol/bandchain/pull/2723) Re-factor to Msg structure
+- (impv) [\#2695](https://github.com/bandprotocol/bandchain/pull/2695) Remove unused images and replace with icon.
```
Comment (evilpeach) [inline CHANGELOG_UNRELEASED.md]: It's duplicated.

File: scan/src/subscriptions/TxSub.re [hunk 3]
```diff
@@ -1250,9 +1343,7 @@ let toExternal =
   timestamp: block.timestamp,
   messages: {
     let msg = messages |> Js.Json.decodeArray |> Belt.Option.getExn |> Belt.List.fromArray;
-    success
-      ? msg->Belt.List.map(Msg.decodeAction)
-      : msg->Belt.List.map(each => Msg.decodeFailAction(each, sender));
+    success ? msg->Belt.List.map(Msg.decodeAction) : msg->Belt.List.map(Msg.decodeFailAction);
```
Comment (evilpeach) [inline scan/src/subscriptions/TxSub.re]: Move ternary inside Belt.List.map

File: scan/src/subscriptions/TxSub.re [hunk 4]
```diff
@@ -910,186 +1029,112 @@ module Msg = {
...
+    | SubmitProposalMsgFail(proposal) => proposal.proposer
+    | DepositMsgSuccess(deposit)
+    | DepositMsgFail(deposit) => deposit.depositor
+    | VoteMsgSuccess(vote)
+    | VoteMsgFail(vote) => vote.voterAddress
+    | WithdrawCommissionMsgSuccess(withdrawal) => withdrawal.validatorAddress
+    | WithdrawCommissionMsgFail(withdrawal) => withdrawal.validatorAddress
+    | MultiSendMsgSuccess(tx)
+    | MultiSendMsgFail(tx) =>
       let firstInput = tx.inputs |> Belt_List.getExn(_, 0);
```
Comment (evilpeach) [inline scan/src/subscriptions/TxSub.re]: Let's move these msg above the IBC and add todo

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — CHANGES_REQUESTED** [review]
1. Reduce margin bottom 
Screen Shot 2563-11-16 at 09 00 02

**Review by evilpeach — APPROVED** [review]

**Pull request merged**